pkgs/applications/misc/insulator2/default.nix

```diff
@@ -20,50 +20,64 @@
 , rustc
 , jq
 , moreutils
+, fetchpatch
 }:
 
 stdenv.mkDerivation rec {
   pname = "insulator2";
-  version = "2.12.2";
+  version = "2.13.2";
 
   src = fetchFromGitHub {
     owner = "andrewinci";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-Bi9GCQr7yox5Plc7o0svRKYi1XoK/HDGj1VbW1z4jac=";
+    hash = "sha256-34JRIB7/x7miReWOxR/m+atjfUiE3XGyh9OBSbMg3m4=";
   };
 
+  patches = [
+    # see: https://github.com/andrewinci/insulator2/pull/733
+    (fetchpatch {
+      name = "fix-rust-1.80.0";
+      url = "https://github.com/andrewinci/insulator2/commit/7dbff0777c4364eec68cf90488d99f06b11dfa98.patch";
+      hash = "sha256-P8rBufYpREP5tOO9vSymvms4f2JbsUEjK7/yn9P7gYk=";
+    })
+  ];
+
   # Yarn *really* wants us to use corepack if this is set
   postPatch = ''
     jq 'del(.packageManager)' package.json | sponge package.json
   '';
 
   yarnOfflineCache = fetchYarnDeps {
     yarnLock = "${src}/yarn.lock";
-    hash = "sha256-ih5NSOvYje981SkVfPHm/u2sS1B36kgxpfe9LmQaxdo=";
+    hash = "sha256-5wOgVrcHJVF07QpnN52d4VWEM3FKw3NdLrZ1goAP2oI=";
   };
 
   cargoDeps = rustPlatform.importCargoLock {
     lockFile = ./Cargo.lock;
     outputHashes = {
-      "apache-avro-0.15.0" = "sha256-bjA/x/IDzAYugsc1vn9fBVKaCiLOJYdA1Q9H2pffBh0=";
-      "openssl-src-111.25.0+1.1.1t" = "sha256-1BEtb38ilJJAw35KW+NOIe1rhxxOPsnz0gA2zJnof8c=";
-      "rdkafka-0.29.0" = "sha256-a739Fc+qjmIrK754GT22Gb/Ftd82lLSUzv53Ej7Khu4=";
+      "apache-avro-0.16.0" = "sha256-v4TeJEhLEqQUgj+EHgFRVUGoLC+SpOUhAXngMP7R7nM=";
       "rust-keystore-0.1.1" = "sha256-Cj64uJFZNxnrplhRuqf9/HK/RAaawzfYHo/J9snZ+TU=";
     };
   };
 
   cargoRoot = "backend/";
-  buildAndTestDir = cargoRoot;
+  buildAndTestSubdir = cargoRoot;
+
+  dontUseCmakeConfigure = true;
+
+  preInstall = ''
+    mkdir -p "$out"
+  '';
 
   nativeBuildInputs = [
+    cmake
     pkg-config
     perl
     rustPlatform.cargoSetupHook
     cargo
     rustc
     cargo-tauri.hook
-    fixup_yarn_lock
```

Comment:
Why is it " migrate to fixup-yarn-lock" while this is deleted?

Reply:
yeah I started like this.
then https://github.com/NixOS/nixpkgs/commit/328e517888f2b56f05f93a11c74b4cabda8107c4 was merged and the build was not working.

So this is more a fix.